pkg/kubelet/dockershim/docker_container.go

```diff
@@ -103,7 +103,6 @@ func (ds *dockerService) CreateContainer(podSandboxID string, config *runtimeApi
 		Name: makeContainerName(sandboxConfig, config),
 		Config: &dockercontainer.Config{
 			// TODO: set User.
-			Hostname:   sandboxConfig.GetHostname(),
```

Comment:
Because it's wrong. We only set the hostname for the infra container (aka the sandbox)